vmm/src/device_manager.rs

```diff
@@ -403,6 +422,52 @@ pub fn get_win_size() -> (u16, u16) {
     (ws.cols, ws.rows)
 }
 
+const TIOCSPTLCK: libc::c_int = 0x4004_5431;
+const TIOCGTPEER: libc::c_int = 0x5441;
+
+pub fn create_pty() -> io::Result<(File, File, PathBuf)> {
+    let main = match OpenOptions::new()
+        .read(true)
+        .write(true)
+        .custom_flags(libc::O_NOCTTY)
+        .open("/dev/pts/ptmx")
+    {
+        Ok(f) => f,
+        _ => OpenOptions::new()
```

Comment:
So I was thinking about this a little and, based on the various pty namespace potential issues, it could exist but not be something the user has permissions to (indicating that maybe it is a mount from a different namespace unrelated to the current one). This fall back seemed the safest in that case as it could potentially be difficult for the user to handle the error otherwise. Thoughts?